repl/src/test/scala/org/apache/spark/repl/ReplSuite.scala

```diff
@@ -382,4 +391,41 @@ class ReplSuite extends SparkFunSuite with BeforeAndAfterAll {
     assertContains(infoLogMessage1, out)
     assertContains(infoLogMessage2, out)
   }
+
+  def runInterpreterAndGetErrors(input: String, readLineDelay: Long): List[String] = {
+    import scala.collection.JavaConverters._
+    import org.apache.log4j.{FileAppender, SimpleLayout}
+
+    val tempFile = Files.createTempFile("repl-test", ".log")
+    tempFile.toFile.deleteOnExit()
+
+    val fileAppender = new FileAppender(new SimpleLayout, tempFile.toFile.getAbsolutePath)
+    fileAppender.setThreshold(Level.ERROR)
+    LogManager.getRootLogger.addAppender(fileAppender)
+    try {
+      runInterpreter("local", input, readLineDelay)
+    } finally {
+      LogManager.getRootLogger.removeAppender(fileAppender)
+    }
+    Files.readAllLines(tempFile).asScala.toList
+  }
+
+  test("inactivity timeout is triggered") {
+    InactivityTimeout.inactivityTimeoutMs = 500
+    InactivityTimeout.isTest = true
```

Comment:
rather than making these `var`s and setting them like this, can you make them confs which you set inside the test?  Then they can be `val`s inside the main code.

If you do set them like this, then they have to be reset back to the original value in a `finally` or `afterEach` etc.

Reply:
Thanks for the review! I have updated the PR and reworked this part to get rid of the vars. Could you please let me know if it looks better now?